test/e2e/test_e2e.py

```diff
@@ -16,14 +16,19 @@ def remove_file(fname):
         pass
 
 
-@pytest.mark.parametrize("fname", [
-    '001_annotations',
-    '002_annotations',
-    '003_annotations',
-    '004_annotations',
-    '005_annotations',
+@pytest.mark.parametrize("fname, subprocess", [
+    ('001_annotations', True),
+    ('002_annotations', True),
+    ('003_annotations', True),
+    ('004_annotations', True),
+    ('005_annotations', True),
+    ('001_annotations', False),
+    ('002_annotations', False),
+    ('003_annotations', False),
+    ('004_annotations', False),
+    ('005_annotations', False)
 ])
-def test_e2e(fname):
+def test_e2e(fname, subprocess):
```

Comment:
why do you need to subprocess=True calls?

Reply:
I suppose I want true "end-to-end" tests from an enduser perspective, who doesn't care wether the program was written in Python or anything else.
Eventually we might swap out the executables underneath but the tests should remain the same (at least the subprocess=True ones). 
It also covers cases where the command line operates unexpectedly different from calling `main` directly. 
Does any of this convince you?

Reply:
Yup. thanks!